### PR TITLE
Add dry-run query param for HealFormat API

### PR DESCRIPTION
**Pull request by krisis**
## Description

This change supports dry-run option in heal-format API, similar to heal-object and heal-bucket.

## Motivation and Context

This allows users to validate the heal format command before executing it for real.
 
## How Has This Been Tested?

Manually tested.
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [  ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: pkg/madmin/examples/heal-format.go [hunk 1]
```diff
@@ -34,13 +34,23 @@ func main() {
 
 	// API requests are secure (HTTPS) if secure=true and insecure (HTTPS) otherwise.
 	// New returns an Minio Admin client object.
-	madmClnt, err := madmin.New("your-minio.example.com:9000", "YOUR-ACCESSKEYID", "YOUR-SECRETACCESSKEY", true)
+	madmClnt, err := madmin.New("localhost:9000", "minio", "minio123", false)
```
Comment (harshavardhana) [inline pkg/madmin/examples/heal-format.go]: any reason to pick this or was it just testing locally?

File: pkg/madmin/examples/heal-format.go [hunk 2]
```diff
@@ -34,13 +34,23 @@ func main() {
...
-	// Heal storage format on available disks.
-	err = madmClnt.HealFormat()
+	// Attempt healing format in dry-run mode.
+	isDryRun := true
+	err = madmClnt.HealFormat(isDryRun)
+	if err != nil {
+		log.Fatalln(err)
+	}
+
+	log.Println("dry run to heal format on available disks was successful.")
```
Comment (harshavardhana) [inline pkg/madmin/examples/heal-format.go]: We don't need this i guess if you wish to keep we need to word this better.

File: cmd/admin-handlers.go [hunk 3]
```diff
@@ -501,7 +502,7 @@ func (adminAPI adminAPIHandlers) HealObjectHandler(w http.ResponseWriter, r *htt
 	writeSuccessResponseHeadersOnly(w)
 }
 
-// HealFormatHandler - POST /?heal
+// HealFormatHandler - POST /?heal&dry-run=[yes|no]
```
Comment (harshavardhana) [inline cmd/admin-handlers.go]: Was thinking that we can avoid making "yes" or no for this. But instead have queryParam itself optional i.e without any value.

This means to heal
```
POST /?heal
``` 

This means fake heal
```
POST /?heal&dry-run
```

Thoughts?

File: pkg/madmin/examples/heal-format.go (hunk 1 above)
Comment (krisis) [reply pkg/madmin/examples/heal-format.go]: It needs to be removed. I used it to test the changes locally.

File: pkg/madmin/examples/heal-format.go (hunk 2 above)
Comment (krisis) [reply pkg/madmin/examples/heal-format.go]: OK

File: cmd/admin-handlers.go (hunk 3 above)
Comment (krisis) [reply cmd/admin-handlers.go]: Ok, I shall make the necessary changes to the other heal APIs that support this too.

**Pull request merged**